### ISSUE TITLE
HackathonPLUS - Breaking News内容可信度评估技术

### ISSUE THREAD

**Issue by xiyu1984** [AlayaNetwork/Developer-Events#10]
## 项目名：
基于隐私计算的内容可信度评估技术

## 项目仓库（可选）：
https://github.com/Hackathon-N-P

## 项目简介：
研究基于隐私计算的内容可信度评估算法，通过内容可信度评估，提高爆料质量，激发社区活力。应用于去中心化的BreakingNews平台，该平台由client端、BreakingNews智能合约组成。
任何参与者可以在BreakingNews平台上发布任何爆料信息，包括文字、图片，其他用户可以对发布的爆料信息进行查看，同时标志支持或反驳，并”跟帖“发表自己的观点（文字、图片）。所有爆料内容，及其相关的观点信息，都会以去中心化的方式记录在案，但仅占用固定规模的链上空间，历史上发生过的任何爆料信息都是不可抵赖的。
所有爆料者都是以匿名的方式参与的，根据爆料者的历史爆料内容，及其爆料反馈（其他参与者对爆料内容的反馈信息），会对每个爆料者的内容可信度进行评估，久而久之，爆料内容越真实，则爆料者可信度越高，其社区的接受程度就越高。

### 里程碑一（1月31日）：
1. 完成系统及关键技术详细设计（11月30日）；
2. 完成适用于资源受限智能合约的内容可信度评估算法研究（12月31日）；
3. 完成算法仿真分析（1月31日）；
4. 完成系统UI设计（1月31日）。

### 里程碑二（3月31日）：
1. 完成系统开发（2月28日），包括：
- 智能合约（2月28日）：
Breaking News发布、评论、点评合约；
用户可信度、内容可信度评估合约；
- Client（2月28日）：
web用户端设计、开发；
2.  完成在PlatON测试网的部署、系统测试，邀请社区试用（3月31日）。

### 里程碑三（里程碑三属于远期计划，全部完成需要结合PlatON隐私AI底层基础设施的上线时间而定）：
1. 基于PlatON隐私计算（可验证计算），优化完善智能合约，通过链上、链下分层架构，实现高效的动态可信度评估机制；
2. 基于PlatON隐私AI基础设施，设计基于深度图神经网络（初步考虑采用GCN模型）的内容可信度评估模型，基于链上、链下分层架构，实现高效的可信度智能评估机制。

## 联系方式：
Email: shanellyy@126.com

LAT地址（PlatON主网LAT账户地址，用于接收奖励）：lat12hhgsx3gla29fze2zygcdkzpgl6mqacuu9lh3p

**Comment by xiyu1984** [comment]
目前里程碑一基本功能都已完成，可以通过http://news.stonelens.com/来试用

算法优化和UI设计正在进行中

**Comment by biganxin** [comment]
Any problems found during the development process can be reported here

**Comment by xiyu1984** [comment]
Hopefully, the WASM contract will also be fully compatible with Ethereum addresses for a more universal use of the Metamask plugin wallet in the PlatON ecosystem.

**Comment by biganxin** [comment]
@benbaley

**Comment by benbaley** [comment]
> ### [issue 1]
> Hopefully, the WASM contract will also be fully compatible with Ethereum addresses for a more universal use of the Metamask plugin wallet in the PlatON ecosystem.

What problems are you encountering now when using metamask or WASM contracts? 
Do you mean to support the bech32 address format inside the contract when writing the WASM contracts?

**Comment by xiyu1984** [comment]
> > ### [issue 1]
> > Hopefully, the WASM contract will also be fully compatible with Ethereum addresses for a more universal use of the Metamask plugin wallet in the PlatON ecosystem.
> 
> What problems are you encountering now when using metamask or WASM contracts? Do you mean to support the EIP55 address format inside the contract when writing the WASM contracts?

Yes, we would like to be able to use the Metamask plugin wallet to interact with the WASM contract. Is it supported now please? If so, what should be done in the WASM contract?

**Comment by benbaley** [comment]
The way calls WASM contract is very different from EVM, so the metamask cannot directly interact with WASM, but it is possible to support both EIP55 and Bech32 in CDT. @fksyuan @jatel

**Comment by xiyu1984** [comment]
# [Additional Milestone]

Added a decentralized review mechanism before content publishing. Content review will be decided by many participants. Multiple reviewers will be selected for each piece of content to be published, and a k/n multi-signature review will be performed. The selection of reviewers will be based on user `credibility` and combined with certain on-chain randomization strategy. Users with higher `credibility` have a higher chance of being selected, and the adoption of review comments (k/n multi-signature approval) will provide a reward to reviewers who share the same opinion.

**Comment by biganxin** [comment]
Issue feedback(500U) rewards have been issued (2021.12.29 Base Price 0.12)
